pkg/apis/mysql/v1alpha1/mysqlcluster_status.go

```diff
@@ -0,0 +1,215 @@
+/*
+Copyright 2018 Pressinfra SRL
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"time"
+
+	// TODO: dont use glog use logutil
+	"github.com/golang/glog"
+
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// UpdateStatusCondition sets the condition to a status.
+// for example Ready condition to True, or False
+func (c *MysqlCluster) UpdateStatusCondition(condType ClusterConditionType,
+	status core.ConditionStatus, reason, msg string) {
+	newCondition := ClusterCondition{
+		Type:    condType,
+		Status:  status,
+		Reason:  reason,
+		Message: msg,
+	}
+
+	t := time.Now()
+
+	if len(c.Status.Conditions) == 0 {
+		glog.V(4).Infof("Setting lastTransitionTime for mysql cluster "+
+			"%q condition %q to %v", c.Name, condType, t)
+		newCondition.LastTransitionTime = metav1.NewTime(t)
+		c.Status.Conditions = []ClusterCondition{newCondition}
+	} else {
+		if i, exist := c.condExists(condType); exist {
+			cond := c.Status.Conditions[i]
+			if cond.Status != newCondition.Status {
+				glog.V(3).Infof("Found status change for mysql cluster "+
+					"%q condition %q: %q -> %q; setting lastTransitionTime to %v",
+					c.Name, condType, cond.Status, status, t)
+				newCondition.LastTransitionTime = metav1.NewTime(t)
+			} else {
+				newCondition.LastTransitionTime = cond.LastTransitionTime
+			}
+			glog.V(4).Infof("Setting lastTransitionTime for mysql cluster "+
+				"%q condition %q to %q", c.Name, condType, status)
+			c.Status.Conditions[i] = newCondition
+		} else {
+			glog.V(4).Infof("Setting new condition for mysql cluster %q, condition %q to %q",
+				c.Name, condType, status)
+			newCondition.LastTransitionTime = metav1.NewTime(t)
+			c.Status.Conditions = append(c.Status.Conditions, newCondition)
+		}
+	}
+}
+
+func (c *MysqlCluster) condExists(ty ClusterConditionType) (int, bool) {
+	for i, cond := range c.Status.Conditions {
+		if cond.Type == ty {
+			return i, true
+		}
+	}
+
+	return 0, false
+}
+
+// GetCondition returns the backup condition of given type
+func (b *MysqlBackup) GetCondition(ty BackupConditionType) *BackupCondition {
+	for _, cond := range b.Status.Conditions {
+		if cond.Type == ty {
+			return &cond
+		}
+	}
+
+	return nil
+}
+
+// UpdateStatusCondition sets the condition to a status.
+// for example Ready condition to True, or False
+func (b *MysqlBackup) UpdateStatusCondition(condType BackupConditionType,
```

Comment:
Should be named `UpdateConditionStatus` because it updates a condition's status and the signature should be on a single line if it fits  in 170 chars.